### ISSUE TITLE
Extend CorsDsl with CorsConfigurationSource property

### ISSUE THREAD

**Issue by IvanPavlov1995** [spring-projects/spring-security#9314]
**Expected Behavior**

`CorsDsl` has
```kotlin
var configurationSource: CorsConfigurationSource? = null
```
similar to `ServerCorsDsl`

**Current Behavior**

There is no possibility to set configuration source for cors dsl configuration

**Context**

I was trying to migrate following code to security dsl:
```kotlin
http.cors().configurationSource { CorsConfiguration().applyPermitDefaultValues() }
```
but I can't set configuration source in dsl:
```kotlin
class SecurityConfig : WebSecurityConfigurerAdapter() {
    override fun configure(http: HttpSecurity) {
        http {
            cors {
                //can't set configuration source here
            }
        }
    }
}
```

**Comment by eleftherias** [comment]
Thanks for reaching out @IvanPavlov1995.

Are you interested in submitting a pull request to add support for the `configurationSource` in the Kotlin DSL?

**Comment by IvanPavlov1995** [comment]
Yes, I would like to try to do that :)

**Comment by eleftherias** [comment]
Awesome @IvanPavlov1995!
Feel free to reach out if you have any questions.

**Comment by eleftherias** [comment]
Closed via f4d78d0
